src/policy/mod.rs

```diff
@@ -162,6 +183,21 @@ mod tests {
         assert_eq!(s.to_lowercase(), output.to_lowercase());
     }
 
+    #[test]
+    fn test_timelock_validity() {
+        // only height
+        assert!(ConcretePol::from_str("after(100)").is_ok());
+        // only time
+        assert!(ConcretePol::from_str("after(1000000000)").is_ok());
+        // disjunction
+        assert!(ConcretePol::from_str("or(after(1000000000),after(100))").is_ok());
+        // conjunction
+        assert!(ConcretePol::from_str("and(after(1000000000),after(100))").is_err());
```

Comment:
Why should this fail?

Reply:
A conjunction of height lock and timelock?

Reply:
Oh, right, I forgot that timelocks are defined implicitly by having a value >500000.

I really don't like this, we shuold modify Miniscript to use a different syntax for them.

Reply:
Any suggestions? For starters, I like the way shesek did it for minsc. https://min.sc/#time-durations